--- a/src/main.py
+++ b/src/main.py
@@ -127,11 +127,9 @@ class RewaitaApplication(Adw.Application):
                 file.write("""
 [Desktop Entry]
 Type=Application
-Name=io.github.swordpuffin.rewaita
-X-XDP-Autostart=io.github.swordpuffin.rewaita
-Exec=flatpak run io.github.swordpuffin.rewaita --background
+Name=Rewaita
+Exec=rewaita --background
 DBusActivatable=true
-X-Flatpak=io.github.swordpuffin.rewaita
                 """
                 )
         else:
