--- a/src/main.py
+++ b/src/main.py
@@ -136,15 +136,16 @@ class RewaitaApplication(Adw.Application):
         success = portal.request_background_finish(result)
         if(success):
             print("Background permission granted")
+            autostart_dir = os.path.join(GLib.getenv("HOME"), ".config", "autostart")
+            if not os.path.exists(autostart_dir):
+                os.makedirs(autostart_dir, exist_ok=True)
             with open(os.path.join(GLib.getenv("HOME"), ".config", "autostart", "rewaita.desktop"), "w") as file:
                 file.write("""
 [Desktop Entry]
 Type=Application
-Name=io.github.swordpuffin.rewaita
-X-XDP-Autostart=io.github.swordpuffin.rewaita
-Exec=flatpak run io.github.swordpuffin.rewaita --background
+Name=Rewaita
+Exec=rewaita --background
 DBusActivatable=true
-X-Flatpak=io.github.swordpuffin.rewaita
                 """
                 )
         else:
